lib/test_broker.dart

```diff
@@ -0,0 +1,147 @@
+import 'package:dsbroker/broker.dart';
+import 'package:dslink/dslink.dart';
+import 'package:dslink_dart_test/test_broker_node_provider.dart';
+import 'dart:convert';
+import 'dart:async';
+import 'dart:io';
+
+class TestBroker {
+  int httpPort;
+  int httpsPort;
+  TestBrokerNodeProvider broker;
+  DsHttpServer server;
+  BrokerDiscoveryClient discovery;
+  SecurityContext context = SecurityContext.defaultContext;
+
+  TestBroker(int httpPort, int httpsPort) {
+    this.httpPort = httpPort;
+    this.httpsPort = httpsPort;
+  }
+
+  Future<Null> start() async {
+    var https = false;
+    var config = JSON.decode(testBrokerConfig);
+
+    dynamic getConfig(String key, [defaultValue]) {
+      if (!config.containsKey(key)) {
+        return defaultValue;
+      }
+      var value = config[key];
+
+      if (value == null) {
+        return defaultValue;
+      }
+
+      return value;
+    }
+
+    updateLogLevel(getConfig("logLevel", "finest"));
+    var downstreamName = getConfig("downstreamName", "downstream");
+    broker = new TestBrokerNodeProvider(downstreamName: downstreamName);
+
+    server = new DsHttpServer.start(
+        getConfig("host", "0.0.0.0"),
+        httpPort: this.httpPort,
+        httpsPort: this.httpsPort,
+        nodeProvider: this.broker,
+        linkManager: this.broker,
+        sslContext: this.context
+    );
+
+    https = getConfig("httpsPort", -1) != -1;
+
+    if (getConfig("broadcast", false)) {
+      var addr = await getNetworkAddress();
+      var scheme = https ? "https" : "http";
+      var port = https ? getConfig("httpsPort") : getConfig("port");
+      var url = getConfig("broadcastUrl", "${scheme}://${addr}:${port}/conn");
```

Comment:
Don't wrap using braces if it's just a variable.

Write it like this `"$scheme://$addr:$port/conn"`